packages/aws-cdk/lib/api/cloudformation-deployments.ts

```diff
@@ -366,6 +360,115 @@ export class CloudFormationDeployments {
+    return Object.values(stackArtifact.template.Resources ?? {}).some((resource: any) => resource.Type === 'AWS::CloudFormation::Stack');
+  }
+
+  private async replaceNestedStacksInParentTemplate(
+    rootStackArtifact: cxapi.CloudFormationStackArtifact, generatedParentTemplate: any, deployedParentTemplate: any,
+    parentStackName: string | undefined, listStackResourcesByStackName: { [key: string] : ListStackResources }, sdk: ISDK,
+  ) {
+    const idResourceMap = Object.entries(generatedParentTemplate.Resources ?? {}).filter(([_, attrs]) => this.isNestedStackResource(attrs));
+    for (const [nestedStackLogicalId, resource] of idResourceMap) {
+      let nestedStackResource = resource as any;
+
+      if (parentStackName && !listStackResourcesByStackName[parentStackName]) {
+        listStackResourcesByStackName[parentStackName] = new LazyListStackResources(sdk, parentStackName);
+      }
+
+      const assetPath = nestedStackResource.Metadata['aws:asset:path'];
+      const nestedStackTemplates = await this.getNestedStackTemplates(
+        rootStackArtifact, assetPath, nestedStackLogicalId, parentStackName, listStackResourcesByStackName,
+      );
+
+      generatedParentTemplate.Resources[nestedStackLogicalId] = nestedStackTemplates.generatedNestedTemplate;
+      generatedParentTemplate.Resources[nestedStackLogicalId].Type = 'AWS::CloudFormation::Stack';
+
+      if (deployedParentTemplate.Resources) {
+        deployedParentTemplate.Resources[nestedStackLogicalId] = nestedStackTemplates.deployedNestedStackTemplate;
+        deployedParentTemplate.Resources[nestedStackLogicalId].Type = 'AWS::CloudFormation::Stack';
+      }
+
+      await this.replaceNestedStacksInParentTemplate(
+        rootStackArtifact, generatedParentTemplate.Resources[nestedStackLogicalId],
+        (deployedParentTemplate.Resources ?? {})[nestedStackLogicalId] ?? {},
+        nestedStackTemplates.nestedStackName, listStackResourcesByStackName, sdk,
+      );
+    }
+  }
+
+  private async getNestedStackTemplates(
+    rootStackArtifact: cxapi.CloudFormationStackArtifact, nestedTemplateAssetPath: string, nestedStackLogicalId: string,
+    parentStackName: string | undefined, listStackResourcesByStackName: { [key:string]: ListStackResources },
+  ): Promise<NestedStackTemplates> {
+    const nestedTemplatePath = path.join(rootStackArtifact.assembly.directory, nestedTemplateAssetPath);
+    // CFN generates the nested stack name in the form `ParentStackName-NestedStackLogicalID-SomeHashWeCan'tCompute,
+    // so we get the ARN and manually extract the name.
+    let evaluateCfnTemplate = undefined;
+    if (parentStackName) {
+      evaluateCfnTemplate = new EvaluateCloudFormationTemplate({
+        stackArtifact: rootStackArtifact,
+        parameters: {},
+        account: '',
+        region: '',
+        partition: '',
+        urlSuffix: () => '',
+        listStackResources: listStackResourcesByStackName[parentStackName],
+      });
+    }
```

Comment:
No need to use `let` here when `const` is easily possible:

```suggestion
    const evaluateCfnTemplate = parentStackName
      ? new EvaluateCloudFormationTemplate({
        stackArtifact: rootStackArtifact,
        parameters: {},
        account: '',
        region: '',
        partition: '',
        urlSuffix: () => '',
        listStackResources: listStackResourcesByStackName[parentStackName],
      })
      : undefined;
```